packages/axios-extension/src/abap/adt-catalog/services/businessobjects-service.ts

```diff
@@ -0,0 +1,89 @@
+import { AdtService } from './adt-service';
+import type { AdtCategory } from '../../types';
+import { XMLParser, XMLValidator } from 'fast-xml-parser';
+
+export type BusinessObject = {
+    name: string;
+    uri: string;
+    description?: string;
+};
+
+/**
+ * BusinessObjectsService implements ADT requests to obtain the list of business objects.
+ */
+export class BusinessObjectsService extends AdtService {
+    /**
+     * @see AdtService.getAdtCatagory()
+     */
+    private static adtCategory = {
+        scheme: 'http://www.sap.com/adt/categories/respository',
+        term: 'search'
+    };
+
+    /**
+     * Get ADT scheme ID.
+     *
+     * @returns AdtCategory
+     */
+    public static getAdtCatagory(): AdtCategory {
+        return BusinessObjectsService.adtCategory;
+    }
+
+    /**
+     * Get all business objects from ADT service.
+     *
+     * @param maxResults - The maximum number of business objects to be returned.
+     * @returns A list of business objects.
+     */
+    public async getBusinessObjects(maxResults?: number): Promise<BusinessObject[]> {
+        const config = {
+            headers: {
+                Accept: 'application/xml'
+            },
+            params: {
+                operation: 'quickSearch',
+                query: `*`,
+                maxResults: maxResults || 10000,
+                objectType: 'BDEF',
+                releaseStatus: 'USE_IN_CLOUD_DEVELOPMENT'
+            }
+        };
+        const response = await this.get('', config);
+        return this.parseBOResponse(response.data);
+    }
+
+    /**
+     * Parse the XML document of business object entries from ADT service.
+     *
+     * @param xml xml document containing business object entries.
+     * @returns A list of business object names.
+     */
+    private parseBOResponse(xml: string): BusinessObject[] {
+        if (XMLValidator.validate(xml) !== true) {
```

Comment:
Would it be possible to reuse the protected `parseResponse` function inherited from `AdtService` for the parsing part of this function? (line 62 to 75)

Reply:
Thanks, done